test/jsonapi_test.exs

```diff
@@ -275,6 +278,70 @@ defmodule JSONAPITest do
     assert Map.has_key?(json, "links")
   end
 
+  describe "with an underscored API" do
+    setup do
+      Application.put_env(:jsonapi, :field_transformation, :underscore)
+
+      on_exit(fn ->
+        Application.delete_env(:jsonapi, :field_transformation)
+      end)
+
+      {:ok, []}
+    end
+
+    test "handles sparse fields properly" do
+      conn =
+        :get
+        |> conn("/posts?include=other_user.company&fields[mytype]=text,excerpt,first_character")
+        |> Plug.Conn.assign(:data, [@default_data])
+        |> MyPostPlug.call([])
+
+      assert %{
+               "data" => [
+                 %{"attributes" => attributes}
+               ]
+             } = Jason.decode!(conn.resp_body)
+
+      assert %{
+               "text" => "Hello",
+               "excerpt" => "He",
+               "first_character" => "H"
+             } == attributes
+    end
+  end
```

Comment:
Last question and I swear I'm done.  Is there a proper test where say `excerpt` is not included in the response b/c it wasn't in the fields map?

Reply:
There is now 😄